run-spill-build/overlaySinglesIntoSpillsSorted.C

```diff
@@ -52,60 +60,114 @@ void overlaySinglesIntoSpillsSorted(std::string inFileName1,
                                     double spillPOT = 5E13,
                                     double spillPeriod_s = 1.2) {
 
-  // get input nu-LAr files
-  TChain* edep_evts_1 = new TChain("EDepSimEvents");
-  edep_evts_1->Add(inFileName1.c_str());
+  // Maximum number of interactions that can be simulated in
+  // one spill in "N Interaction" mode. Choice of this number
+  // here is somewhat arbitrary, seemed like a very safe
+  // upper limit on the number of nu-LAr events we'd want to 
+  // simulate ever in a single spill (for stress testing).
+  int n_int_max = 10000;
+  // Check that we are considering, nu-LAr only, nu-rock only or both.
+  if (inFile1POT==0. && inFile2POT==0.) {
+    throw std::invalid_argument("nu-LAr POT and nu-rock POT cannot both be zero!");
+  }
+  // "N Interaction" mode only supported in nu-LAr mode.
+  else if (spillPOT <= (double)n_int_max && inFile2POT>0.) {
+    throw std::invalid_argument("N Interaction mode does not support nu-rock POT input");
+  }
 
-  // get input nu-Rock files
-  TChain* edep_evts_2 = new TChain("EDepSimEvents");
-  edep_evts_2->Add(inFileName2.c_str());
+  // Useful bools for keeping track of event types being considered.
+  bool have_nu_lar = false;
+  bool have_nu_rock = false;
+  bool is_n_int_mode = false;
 
   // get input nu-LAr files
+  TChain* edep_evts_1 = new TChain("EDepSimEvents");
   TChain* genie_evts_1 = new TChain("DetSimPassThru/gRooTracker");
-  genie_evts_1->Add(inFileName1.c_str());
+  if(inFile1POT > 0.) {
+    edep_evts_1->Add(inFileName1.c_str());
+    genie_evts_1->Add(inFileName1.c_str());
+    have_nu_lar = true;
+    if(spillPOT <= (double)n_int_max) is_n_int_mode = true;
+  }
+  else std::cout << "nu-rock file POT stated to be zero, spills will be LAr only" << std::endl;
   gRooTracker genie_evts_1_data(genie_evts_1);
 
   // get input nu-Rock files
+  TChain* edep_evts_2 = new TChain("EDepSimEvents");
   TChain* genie_evts_2 = new TChain("DetSimPassThru/gRooTracker");
-  genie_evts_2->Add(inFileName2.c_str());
+  if(inFile2POT > 0.) {
+    edep_evts_2->Add(inFileName2.c_str());
+    genie_evts_2->Add(inFileName2.c_str());
+    have_nu_rock = true;
+  }
+  else std::cout << "nu-LAr file POT stated to be zero, spills will be rock only" << std::endl;
```

Comment:
Swap?

Reply:
Should we throw an error if `inFile2POT >0 && is_n_int_mode`?

Reply:
As above re. the swap.

With regard to your second comment this should be covered by [this logic](https://github.com/DUNE/2x2_sim/blob/339348f2252266d51be93bd18327859175ef51c3/run-spill-build/overlaySinglesIntoSpillsSorted.C#L74).